src/lint/mod.rs

```diff
@@ -77,6 +85,32 @@ pub struct LintCfg {
     pub annotate_lints: bool,
 }
 
+impl LintCfg {
+    pub fn try_deserialize(
+        val: &serde_json::Value,
+        unknowns: &mut Vec<String>,
+    ) -> Result<LintCfg, String> {
+        // Handle unknown fields by collecting field names
+        if let Some(obj) = val.as_object() {
+            let known_fields = [
+                "sp_brace", "sp_punct", "nsp_funpar", "nsp_inparen", 
+                "nsp_unary", "nsp_trailing", "long_lines", "indent_size", 
+                "indent_no_tabs", "indent_code_block", "indent_closing_brace", 
+                "indent_paren_expr", "indent_switch_case", "indent_empty_loop", 
+                "annotate_lints"
+            ];
```

Comment:
This should be automated with some macro maybe, so that this list does not need maintaining when rules are added or names are changed.

Reply:
Indeed. You can leverage the derived deserialize and serde_ignored to accomplish this. (again, the strategy used for config deserialization works nicely. if you are not concerned with deprecated/duplicated options, those parts can be left out)